sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -562,20 +575,26 @@ class FilesElement extends Polymer.Element {
         .then((closeResult: InputDialogCloseResult) => {
           if (closeResult.confirmed && closeResult.userInput) {
             const newName = this.currentPath + '/' + closeResult.userInput;
+
+            let success: boolean;
             return ApiManager.renameItem(selectedObject.path, newName)
-              .then(() => this._fetchFileList())
               .then(() => {
+                success = true;
                 // Dispatch a success notification
                 const message = 'Renamed ' + selectedObject.name +
                     ' to ' + closeResult.userInput + '.';
                 this.dispatchEvent(new NotificationEvent(message));
-              });
-              // TODO: [yebrahim] Re-select the renamed item after refresh
+              })
+              .catch((e: Error) => {
+                success = false;
+                Utils.showErrorDialog('Error renaming item', e.message);
+              })
+                // Only refresh the file list on success
+              .then(() => success === true ? this._fetchFileList() : null);
```

Comment:
It feels wrong to me to be using a success flag when the Promise mechanism is supposed to have success and failure control flow built in by using then and catch.
What if you add a throw (or return Promise.reject()) at the end of your catch block so that the following then() does not get executed? If you don't catch that elsewhere, does it bubble out as an error somewhere, or will it just be swallowed?
I'll approve, in case you still think the current code is the best way of doing it, but would be happier with a solution that uses then and catch to manage control flow.